src/Microsoft.Azure.WebJobs.Host/Triggers/TriggeredFunctionBinding.cs

```diff
@@ -56,7 +58,7 @@ internal class TriggeredFunctionBinding<TTriggerValue> : ITriggeredFunctionBindi
             {
                 ITriggerData triggerData = await _triggerBinding.BindAsync(value, context);
                 triggerProvider = triggerData.ValueProvider;
-                bindingData = triggerData.BindingData;
+                bindingData = triggerData.BindingData.Concat(AppSettingsBinding.CreateBindingData()).ToDictionary(x => x.Key, x => x.Value);
```

Comment:
Hey, I just realized that our existing JobHostConfiguration.INameResolver mechanism would satisfy your requirements. Basically, you provide your own INameResolver impl. that maps names to values (e.g. you can read from app settings, or wherever you want). Then in SingletonAttribute you can use an expression like [Singleton("%mysetting%")]. During the binding process, all our exising attributes that support binding expressions will use INameResolver to resolve any %% delimited values.

Issue was that Singleton didn't support this, but I just fixed that here: https://github.com/Azure/azure-webjobs-sdk/commit/1b9c21e593220b25945a371ab17733afb6f60e5f

:)